pre_commit/repository.py

```diff
@@ -64,11 +64,18 @@ def languages(self):
 
     @cached_property
     def additional_dependencies(self):
-        dep_dict = defaultdict(lambda: defaultdict(set))
+        dep_dict = defaultdict(lambda: defaultdict(list))
         for _, hook in self.hooks:
-            dep_dict[hook['language']][hook['language_version']].update(
-                hook.get('additional_dependencies', []),
-            )
+            for dep in hook.get('additional_dependencies', []):
+                if dep not in \
+                        dep_dict[hook['language']][hook['language_version']]:
+                    dep_dict[hook['language']][hook['language_version']].\
+                        append(dep)
+                else:
+                    logger.warning(
+                        'Additional dependency {} is listed twice for hook '
+                        '{}'.format(dep, hook['name'])
+                    )
```

Comment:
A warning should not be raised here.

This is a perfectly valid configuration:

```yaml
    hooks:
    -   id: puppet-validate
        language_version: '2.1.5'
        additional_dependencies: ['puppet:4.5.3']
    -   id: epp-validate
        language_version: '2.1.5'
        additional_dependencies: ['puppet:4.5.3']
```